src/solver/conv_asm_1x1u.cpp

```diff
@@ -243,75 +244,182 @@ bool PerformanceConfigConvAsm1x1U::operator==(const PerformanceConfigConvAsm1x1U
 
 bool PerformanceConfigConvAsm1x1U::IsValidValue() const
 {
-    // clang-format off
-    return IsLinear<1,4>(read_size)
-        && Is_1_4_8_12_to_32(k_mult)
-        && IsLinear<1,16>(chunks_per_wave)
-        && IsTwoPower<1,64>(chunk_size)
-        && IsLinear<1,8>(n_mult)
-        && IsTwoPower<1,32>(c_mult)
-        && IsLinear<1,8>(waves_c_in_group)
-        && IsTwoPower<1,8>(waves_k_in_group); // clang-format on
+    bool is_valid_value = true;
+    if(read_size != -1)
+    {
+        is_valid_value = is_valid_value && IsLinear<1, 4>(read_size);
```

Comment:
Why not do
```
if (!IsLinear<1, 4>(read_size))
    return false;
```

And if all checks pass, `return true;` at the end?

Reply:
Yeah sure, that would look cleaner.